cellprofiler_core/readers/gcs_reader.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+from urllib.parse import urlparse
+
+import imageio
+from cellprofiler_core.readers.imageio_reader import ImageIOReader
+from google.cloud import storage
+
+from ..reader import Reader
+
+SUPPORTED_EXTENSIONS = {'.tiff'}
+
+
+class GcsReader(ImageIOReader):
+  """ Reads images from bucket(s) in Google Cloud Storage via user's Application Default Credential (ADC).
+  Leverages ImageIOReader image processing methods.
+
+  Prerequisites:
+  1. User has authenticated with Google Cloud Storage Application Default Credential (ADC) by running command, `gcloud auth application-default login`.
+  2. User has validated that the Google Cloud Storage project to which the LoadData module URLs point is
+  the current [default] project by running `gcloud config list`.
```

Comment:
This prerequisite is not quite correct. I think this prerequisite can just be removed from the list.

People can use the Python GCS client to read data in projects for which they are not members, such as public data in GCS buckets.

Reply:
Good point -- I have removed that item from the comment as it isn't strictly necessary.